### PR TITLE
Add associated method to DecodeLimit for advancing input index

### PR DESCRIPTION
**Pull request by KiChjang**

### CODE REVIEW

File: src/depth_limit.rs [hunk 1]
```diff
@@ -82,6 +87,15 @@ impl<T: Decode> DecodeLimit for T {
...
+	fn decode_and_advance_with_depth_limit<I: Input>(limit: u32, input: &mut I) -> Result<Self, Error> {
+		let mut input = DepthTrackingInput {
+			input,
+			depth: 0,
+			max_depth: limit,
+		};
+		T::decode(&mut input)
+	}
+
 	fn decode_with_depth_limit(limit: u32, input: &[u8]) -> Result<Self, Error> {
```
Comment (shawntabrizi) [inline src/depth_limit.rs:99]: I think we need to put a deprecated on this

Comment (bkchr) [reply src/depth_limit.rs:99]: Not really. However, we should create an issue to merge both of these methods in the next major release. @KiChjang please also create some new milestone for the next major and attach the issue.

**Review by bkchr — APPROVED** [review]

**Review by arkpar — APPROVED** [review]

File: src/depth_limit.rs (hunk 1 above)
Comment (KiChjang) [reply src/depth_limit.rs:99]: Filed as #287 and added a SCALE codec 3.0 milestone.

**Comment by KiChjang** [comment]
Unsure what to do about the status checks? The CI failures seem unrelated to this PR.

**Comment by thiolliere** [comment]
if errors are legit then we should fix master

**Comment by KiChjang** [comment]
This may need to be fast tracked to a new minor version to solve a security bug; it can't wait for 3.0.

File: src/bit_vec.rs [hunk 2]
```diff
@@ -211,6 +211,10 @@ mod tests {
 	}
 
 	#[test]
+	// Flacky test due to:
+	// * https://github.com/bitvecto-rs/bitvec/issues/135
+	// * https://github.com/rust-lang/miri/issues/1866
+	#[cfg(not(miri))]
```
Comment (thiolliere) [inline src/bit_vec.rs:217]: CI seems to fail due to this flacky test

**Review by thiolliere — APPROVED** [review]

**Comment by thiolliere** [comment]
> This may need to be fast tracked to a new minor version to solve a security bug; it can't wait for 3.0.

yes we can bump to a new minor in this PR or just create a new PR to bump just after.

Also it needs an entry in the CHANGELOG

**Pull request merged**